File: Python/383. Ransom Note/lc383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        rdict,mdict={},{}
        for r in ransomNote:
            rdict[r]=rdict.get(r,0)+1
        for m in magazine:
            mdict[m]=mdict.get(m,0)+1
        for k,v in rdict.items():
            if v>mdict.get(k,0):
                return False
        return True

File: C++/383. Ransom Note/lc383.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map <char, int> rMap,mMap;
        for (auto r: ransomNote)
        {
            rMap[r]++;
        }
        for (auto m: magazine)
        {
            mMap[m]++;
        }
        for (auto it=rMap.begin();it!=rMap.end();it++)
        {
            if(it->second >mMap[it->first])
                return false;
        }
        return true;
        
    }
};
